source/kraken/configuration.cpp

```diff
@@ -118,8 +119,12 @@ boost::optional<std::string> Configuration::chaos_database() const{
     }
     return result;
 }
-int Configuration::nb_thread() const{
-    return this->vm["GENERAL.nb_threads"].as<int>();
+int Configuration::nb_threads() const{
+    int nb_threads = vm["GENERAL.nb_threads"].as<int>();
+    if (nb_threads < 0) {
+        throw std::invalid_argument("nb_threads cannot be negative");
+    }
+    return size_t(nb_threads);
```

Comment:
sorry for that, we have to remove cast to be consistent (how we use nb_threads() now is safe)
and as it was not the point of the PR I didn't want to refactor the int on all use